lib/spack/spack/build_systems/oneapi.py

```diff
@@ -109,18 +109,28 @@ def install_component(self, installer_path):
             # Installer checks $XDG_RUNTIME_DIR/.bootstrapper_lock_file as well
             bash.add_default_env("XDG_RUNTIME_DIR", join_path(self.stage.path, "runtime"))
 
-            bash(
-                installer_path,
-                "-s",
-                "-a",
-                "-s",
-                "--action",
-                "install",
-                "--eula",
-                "accept",
-                "--install-dir",
-                self.prefix,
-            )
+            if is_nvidia:
```

Comment:
I think you can put this code directly in the compiler installer since it is not needed by any of the other oneapi installers.